battle.go

```diff
@@ -413,18 +413,29 @@ func (b *Battle) SimulateRound() ([]Transaction, bool) {
 				})
 				// Handle draining moves (Absorb, Mega Drain, Giga Drain, Drain Punch, etc.)
 				if move.Drain() != 0 {
-					drain := damage * uint(move.Drain()) / 100
-					if user.HeldItem == ItemBigRoot {
-						drain = drain * 130 / 100 // 30% more HP than normal
+					drain := int(damage) * move.Drain() / 100
+					if drain > 0 {
+						// add damage dealt to health
+						if user.HeldItem == ItemBigRoot {
+							drain = drain * 130 / 100 // 30% more HP than normal
+						}
 					}
 					if drain == 0 {
 						// Min 1 HP drain
 						drain = 1
 					}
```

Comment:
This pattern of setting min damage to 1 keeps repeating. I think we could benefit from having the minimum be applied in the `DamagTransaction` /`HealTransaction` and either:
- Set a flag in the transaction that no damage should be dealt
- Replace 0 damage transactions with `MoveFailTransaction` with `ReasonImmune` or similar